src/key-managment/AwsKeyManagement.cs

```diff
@@ -0,0 +1,107 @@
+
+        }
+
+        public async Task<string> Decrypt(string encryptedData, string serviceAccountId)
+        {
+            var encryptedEncryptionKey = encryptedData.Split('$')[1];
+            var actualEncryptedData = encryptedData.Split('$')[2];
+            
+            var decryptionResult = await mAmazonKeyManagementService.DecryptAsync(new DecryptRequest
+            {
+                CiphertextBlob = new MemoryStream(Convert.FromBase64String(encryptedEncryptionKey)),
+            });
+
+            var encryptionKey = await ConvertMemoryStreamToBase64String(decryptionResult.Plaintext);
+                
+            mSymmetricKeyManagement.SetEncryptionKey(encryptionKey);
+            return await mSymmetricKeyManagement.Decrypt(actualEncryptedData, serviceAccountId);
+        }
+
+        private static async Task<string> ConvertMemoryStreamToBase64String(MemoryStream ms)
+        {
+            return Convert.ToBase64String(ms.ToArray());
+        }
+
+        private async Task<(MemoryStream encryptionKey, string encryptedEncryptionKey)> GenerateEncryptionKey(string keyAlias)
+        {
+            GenerateDataKeyResponse generateKeyResponse = null;
+            try
+            {
+                generateKeyResponse = await mAmazonKeyManagementService.GenerateDataKeyAsync(new GenerateDataKeyRequest { KeyId = keyAlias, KeySpec = "AES_256"});
+
+            }
+            catch (NotFoundException e)
+            {
+                await GenerateMasterKey(keyAlias);
+                generateKeyResponse = await mAmazonKeyManagementService.GenerateDataKeyAsync(new GenerateDataKeyRequest { KeyId = keyAlias, KeySpec = "AES_256"});
+            }
+
+            if (generateKeyResponse.HttpStatusCode != HttpStatusCode.OK)
+            {
+                throw new Exception($"Couldn't generate encryption key for {keyAlias}");
+            }
+
+            return (generateKeyResponse.Plaintext, await ConvertMemoryStreamToBase64String(generateKeyResponse.CiphertextBlob));
+        }
+
+        private async Task GenerateMasterKey(string keyAlias)
+        {
+            String policy = "{" +
+                            "  \"Version\": \"2012-10-17\"," +
+                            "  \"Statement\": [{" +
+                            "    \"Sid\": \"Allow access for KamusUser\"," +
+                            "    \"Effect\": \"Allow\"," +
+                            "    \"Principal\": {\"AWS\": \""+mUserArn+"\"}," +
+                            "    \"Action\": [" +
+                            "      \"kms:Encrypt\"," +
+                            "      \"kms:Describe*\"," +
+                            "      \"kms:Get*\"," +
+                            "      \"kms:List*\"," +
+                            "      \"kms:GenerateDataKey*\"," +
+                            "      \"kms:Decrypt\"," +
+                            "      \"kms:Delete*\"," +
+                            "      \"kms:CreateAlias\"" +
```

Comment:
Do we need create alias?

Reply:
This code doesn't exist anymore, I think we can start with default key policies.